src/utils/vc/credentials.js

```diff
@@ -27,7 +30,39 @@ function getId(obj) {
   if (typeof obj === 'string') {
     return obj;
   }
-  return obj.id;// || 'did:dock:' + String(obj);
+  return obj.id;
+}
+
+export function formatToJWTPayload(keyDoc, cred) {
+  const kid = keyDoc.id;
+  const credentialIssuer = cred.issuer;
+  const subject = cred.credentialSubject.id;
+  const validFrom = cred.validFrom || cred.issuanceDate;
+  const { issuanceDate, expirationDate } = cred;
+
+  // Reference: https://www.w3.org/TR/vc-data-model/#jwt-encoding
+  const vcJwtPayload = {
+    jti: cred.id,
+    sub: subject || '',
+    iss: credentialIssuer.id || credentialIssuer,
+    iat: Math.floor(Date.parse(issuanceDate) / 1000),
+    vc: cred,
+  };
+
+  if (validFrom) {
+    vcJwtPayload.nbf = Math.floor(Date.parse(validFrom) / 1000);
```

Comment:
The reference link above says to construct `nbf` from `issuanceDate` and not `validFrom`

Reply:
`validFrom` is `issuanceDate` if not supplied - please read VC data model disclaimer:
```
It is expected that the next version of this specification will add the validFrom [property](https://www.w3.org/TR/vc-data-model/#dfn-property) and will deprecate the issuanceDate [property](https://www.w3.org/TR/vc-data-model/#dfn-property) in favor of a new issued [property](https://www.w3.org/TR/vc-data-model/#dfn-property). The range of values for both properties are expected to remain as [[XMLSCHEMA11-2](https://www.w3.org/TR/xmlschema11-2/#dateTime)] combined date-time strings. Implementers are advised that the validFrom and issued [properties](https://www.w3.org/TR/vc-data-model/#dfn-property) are reserved and use for any other purpose is discouraged.
```

just trying to future proof

Reply:
Can you add this note in the code?

Reply:
updated